app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala

```diff
@@ -49,29 +54,54 @@ class MessagesController @Inject() (appConfig: AppConfig, cc: ControllerComponen
   private val UUID_PATTERN         = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$".r
   private val BEARER_TOKEN_PATTERN = "^Bearer (\\S+)$".r
 
-  def post(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
+  def routeToEIScheck(client: String, customsOffice: String): Boolean = appConfig.clientAllowList.contains(
+    client
+  ) && appConfig.enableProxyMode && !customsOffice.isBlank && (customsOffice == "gb" | customsOffice == "xi")
+
+  def post(customsOffice: String): Action[Source[ByteString, _]] = Action.async(streamFromMemory) {
```

Comment:
Can we send a request without a customs office? If not, we should probably allow it still by making this `customsOffice` an `Option` and having two routes, one with the customs code in the URL, one without, and using this same method for both endpoints.